### PR TITLE
[ML] adding running_state to datafeed stats object

### PR DESCRIPTION
**Pull request by benwtrent**
It is useful to know the following information when reading datafeed stats:

 - Is the datafeed a "real-time" datafeed, i.e. a datafeed without a configured `end` time
 - Has the datafeed processed all past data available at the time of starting.

This object is only available if the datafeed task has been created.

It has the form:

```
"running_state": {
  "real_time_configured": <boolean>,
  "real_time_running": <boolean>
}
```

### CODE REVIEW

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc [hunk 1]
```diff
@@ -103,6 +103,24 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=node-transport-address]
...
++
+--
+[%collapsible%open]
+====
+`is_real_time`:::
+(boolean) Indicates if the {dfeed} is "real-time"; meaning that the {dfeed}
+has no configured `end` time.
+
+`finished_look_back`:::
+(boolean) Has the {dfeed} finished running on the available past data. For {dfeeds}
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:118]:
```suggestion
(boolean) Indicates whether the {dfeed} has finished running on the available past data. For {dfeeds}
```

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc [hunk 2]
```diff
@@ -103,6 +103,24 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=node-transport-address]
...
+that without a configured `end` time, this means that the {dfeed} is now running on
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:119]:
```suggestion
without a configured `end` time, this means that the {dfeed} is now running on
```

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc [hunk 3]
```diff
@@ -103,6 +103,24 @@ include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=node-transport-address]
...
+provided if the {dfeed} is started.
++
+--
+[%collapsible%open]
+====
+`is_real_time`:::
+(boolean) Indicates if the {dfeed} is "real-time"; meaning that the {dfeed}
+has no configured `end` time.
+
+`finished_look_back`:::
```
Comment (droberts195) [inline docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:117]: Our existing docs don't use the term "lookback".  I am not sure we should introduce it now.

One solution would be to rename `is_real_time` to `real_time_configured` and `finished_look_back` to `real_time_running`.  Then that doesn't involve introducing a new public term.

**Review by droberts195 — COMMENTED** [review]
Looks good.

My only issue is whether we introduce the term "lookback" to the world.  I am leaning towards not doing this.

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc (hunk 3 above)
Comment (benwtrent) [reply docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:117]: @droberts195 

While its true that we don't use it in docs, `lookback` is used in audit messages:

```
    public static final String JOB_AUDIT_DATAFEED_LOOKBACK_COMPLETED = "Datafeed lookback completed";
    public static final String JOB_AUDIT_DATAFEED_LOOKBACK_NO_DATA = "Datafeed lookback retrieved no data";
```

Comment (benwtrent) [reply docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:117]: I am happy to change to both to indicate `real_time`. But we do use lookback in user facing things.

Comment (droberts195) [reply docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:117]: > `lookback` is used in audit messages

Oh, interesting.  I think we made the decision in 2016 not to use it anywhere user-facing, but those audit messages were added in 2017, by which time the 2016 decision had been forgotten.  I guess the lesson is that we should change our terminology in internal code as well as what's immediately user-facing to stop internal terminology leaking out later on.

I still think there is a benefit in not propagating the term to fields that will appear in every high level client's public API.  At the moment we could change the wording of those audit messages without making a breaking change, but the REST responses are harder to change once published.

Comment (benwtrent) [reply docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:117]: okey dokey, I will rename the variables. 

> I guess the lesson is that we should change our terminology in internal code as well as what's immediately user-facing to stop internal terminology leaking out later on.

100%. Renaming things everywhere prevents this sort of thing.

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**